Arrowgene.Ddon.Database/DdonDatabaseBuilder.cs

```diff
@@ -53,5 +65,29 @@ public static DdonSqLiteDb BuildSqLite(string sqLiteFolder, string sqLitePath, b
 
             return db;
         }
+
+        public static DdonPostgresDb BuildPostgres(DatabaseSetting settings)
+        {
+            DdonPostgresDb db = new DdonPostgresDb(settings);
+            if (db.CreateDatabase())
+            {
+                ScriptRunner scriptRunner = new ScriptRunner(db);
+                scriptRunner.Run(Path.Combine(settings.SqLiteFolder, "Script/schema_postgres.sql"));
```

Comment:
Maintaining multiple schemas for different DBMS could slow down development, or make it so one DBMS is better supported than the others. Is it not possible to have a single schema file using common SQL without using vendor specific extensions?

Reply:
Unfortunately it's not just about vendor specific extensions but rather differing engine behavior.
- MariaDB
	- Behavioral differences at schema level:
		- Does not support TEXT primary keys properly, at most 768 characters can be indexed -> https://mariadb.com/kb/en/text/
		- Does not support TEXT foreign keys -> https://mariadb.com/kb/en/foreign-keys/
- PostgreSQL
	- Syntax differences at schema level
		- TIMESTAMP (WITH/WITHOUT TIME ZONE) = DATETIME
		- BYTEA = BLOB

To properly tackle the MariaDB-related issues would require changes to the underlying data model.

Reply:
Got rid of TEXT and replaced it with VARCHAR 8 for item UID (according to Item.cs this is the max length) & VARCHAR 32 for settings key (this seems currently unused, unsure what length will be necessary).
Added custom data types (domains) for PSQL, this at least allows for carrying over schema definitions more easily from the other engines.

Reply:
Another idea: We get rid of PSQL and favor MariaDb, as the syntax is closer to that of SQLite in some places. As long as one "proper" RDBMS is supported for production purposes we don't need multiple.

Reply:
If timestamp data types are an issue we can always store the string representation, or maybe better store the unix time as a number instead so we can still compare and operate with them in queries.
Regarding BLOB, from what i'm reading, PSQL supports BLOB as its part of the SQL standard but favours BYTEA instead. We can probably still work with BLOB
https://www.postgresql.org/docs/current/datatype-binary.html

Reply:
The documentation is a bit confusing, but BLOB is not recognized as a type & throws an error when attempting to be used.

Reply:
For now I have added a schema syntax transformation in the database builder to adapt the small differences:
e.g.
```
        public static DdonPostgresDb BuildPostgres(string databaseFolder, string host, string user, string password, string database, bool wipeOnStartup)
        {
            DdonPostgresDb db = new DdonPostgresDb(host, user, password, database, wipeOnStartup);
            if (db.CreateDatabase())
            {
                string schemaFilePath = Path.Combine(databaseFolder, "Script/schema_sqlite.sql");
                String schema = File.ReadAllText(schemaFilePath, Encoding.UTF8);
                schema = schema.Replace(" DATETIME ", " TIMESTAMP WITH TIME ZONE ");
                schema = schema.Replace(" INTEGER PRIMARY KEY AUTOINCREMENT ", " SERIAL PRIMARY KEY ");
                schema = schema.Replace(" BLOB ", " BYTEA ");
                File.WriteAllText(schemaFilePath, schema);
                
                ScriptRunner scriptRunner = new ScriptRunner(db);
                scriptRunner.Run(Path.Combine(databaseFolder, "Script/schema_postgres.sql"));
            }

            return db;
        }
```

Reply:
Does it work replacing BLOB with BYTEA? The documentation mentions that the input format is different.
We should be careful with string replacements like that, it's its own entire can of worms, there could be a line with no whitespace surrounding the data type

Reply:
The driver handles this low-level stuff, we are just storing storage sorts (byte arrays) as blobs. As long as you don't attempt to copy data from one db to another I don't think this will ever come up as an issue.

Regarding string replacements: It's either that or separate files or not providing any schema support and leaving it up to "3rd parties" in a bring your own schema kind of way. Unless you have some other ideas. I have checked the SQLite syntax (https://www.sqlite.org/syntax/create-table-stmt.html) and adapted it via regexes to be more robust.

Reply:
Regex seems like a good enough compromise without having to build an entire lexical analyser for this